csrc/scheduler/utils.cpp

```diff
@@ -170,21 +170,32 @@ std::optional<size_t> mergeDims(
   if (to_merge.size() == 1) {
     return to_merge[0];
   }
-  std::sort(to_merge.begin(), to_merge.end());
```

Comment:
hmmm. these changes are unintentionally. I'll put this in the scheduler change instead.